### PR TITLE
abort all processes when one fails

### PR DESCRIPTION
**Pull request by kentcdodds**
ref #48

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -18,40 +19,106 @@ function runPackageScripts({scriptConfig, scripts, args, options = {}}, callback
...
   }
   const scriptNames = arrify(scripts)
-  const method = options.parallel ? 'map' : 'mapSeries'
-  async[method](scriptNames, (scriptName, cb) => {
-    const child = runPackageScript({scriptConfig, options, scriptName, args})
-    if (child.on) {
-      child.on('exit', exitCode => cb(null, exitCode))
-    } else {
-      cb(child)
+  if (options.parallel) {
```
Comment (kentcdodds) [inline src/index.js]: I feel like the code below could be greatly simplified with RxJS. But I have no idea how to use Observables. Someone please help me!

Comment (SamVerschueren) [inline src/index.js]: I don't really see an opportunity for observables. Might be missing something though. Why do you think observables make sense here?

Comment (kentcdodds) [inline src/index.js]: I've never used Observables, but I thought that they're supposed to really help with complex async stuff. :shrug:

Comment (SamVerschueren) [inline src/index.js]: I think it can be done a little bit cleaner though. I'll  do a PR later today or this weekend.

Comment (ta2edchimp) [inline src/index.js]: Maybe [`zip`](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/zip.md) is what you're looking for (re RxJS)? Merge multiple Observables/Promises, subscribe to the result and have ha single handling of their results ...

Comment (kentcdodds) [inline src/index.js]: I don't know what I'm looking for.. I just think this code could be cleaned up a bit...

Comment (SamVerschueren) [inline src/index.js]: When running in parallel, you can just use `Promise.all`, when executing serially you can do something like this.

```js
const result = promises.reduce((res, task) => {
   return res.then(() => task());
});

result.then(...);
```

**Comment by kentcdodds** [comment]
Sweet I think this is good to :shipit: 

Could I get some people to test it? I've released it as the beta: `npm install p-s@beta`. Just make sure your stuff's not broken?

cc @boneskull, @tleunen, @nkbt, @DavidWells, @Hypercubed, @rowanoulton 

This is a significant refactor and I'd hate to break anyone due to this bug fix! (Ref #48)

**Comment by kentcdodds** [comment]
:shipit:

**Comment by kentcdodds** [comment]
This is good to merge by anyone as soon as it's reviewed!

File: src/index.js [hunk 2]
```diff
@@ -1,57 +1,125 @@
-import spawn from 'spawn-command'
-import async from 'async'
+import spawn from 'spawn-command-with-kill'
+import Promise from 'bluebird'
 import colors from 'colors/safe'
-import {isString, find, clone} from 'lodash'
+import {isString, clone} from 'lodash'
```
Comment (SamVerschueren) [inline src/index.js]: Full lodash is quite huge, why not install them separately?

**Comment by Hypercubed** [comment]
I gave it a quick run on OSX and windows.  No issues other than an unrelated issue on windows (where you aware that the command `package-scripts` fails on windows?).

So... LGTM.

**Comment by kentcdodds** [comment]
> where you aware that the command `package-scripts` fails on windows?

Oh no! I wasn't. Could you file an issue for that? Thanks for testing it out @Hypercubed!

**Comment by kentcdodds** [comment]
:shipit:

File: src/index.js (hunk 2 above)
Comment (kentcdodds) [inline src/index.js]: I get you, but honestly I'm using about a dozen methods and I'd rather avoid installing each separately/managing them. It's 1.4MB (uncompressed) and the likelihood that someone's already depending on lodash in their project (even if it's transitively) is quite high.

Thanks for the suggestion.

**Comment by boneskull** [comment]
haha, I need more than 22h notice :wink:

**Comment by kentcdodds** [comment]
I guess I was anxious and excited :-)

**Comment by boneskull** [comment]
Have you considered therapy? :wink:

**Pull request merged**